File: Solution/0242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        frequent_words = {}
        
        for s_str in s:
            if s_str not in frequent_words.keys():
                frequent_words[s_str] = 0
            frequent_words[s_str] += 1
        
        for t_str in t:
            if t_str not in frequent_words.keys():
                frequent_words[t_str] = 0
            frequent_words[t_str] -= 1
        
        for value in frequent_words.values():
            if value !=0:
                return False
        
        return True

File: Solution/0242.cpp
class Solution {
    public:
        bool isAnagram(string s, string t) {
            int s_length = s.length(), t_length = t.length();
            if (s_length != t_length) return false;
            unordered_map<char, int> count;
    
            for (int i = 0; i < s_length; ++i) {
                auto it = count.find(s[i]);
                if (it == count.end()) 
                    ++count[s[i]];
            }
    
            for (int i = 0; i < t_length; ++i) {
                auto it = count.find(t[i]);
                if (it == count.end()) 
                    return false;
                else {
                    if (--it->second == 0) count.erase(t[i]);
                }
            }
            return count.empty();
        }
    };